data-prepper-plugins/s3-source/src/integrationTest/java/com/amazon/dataprepper/plugins/source/SqsWorkerIT.java

```diff
@@ -0,0 +1,116 @@
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class SqsWorkerIT {
+    private S3Client s3Client;
+    private SqsClient sqsClient;
+    private S3Service s3Service;
+    private Buffer<Record<Event>> buffer;
+    private S3SourceConfig s3SourceConfig;
+    private S3ObjectGenerator s3ObjectGenerator;
+    private String bucket;
+    private Codec codec;
+    private PluginMetrics pluginMetrics;
+
+    @BeforeEach
+    void setUp() {
+        s3Client = S3Client.builder()
+                .region(Region.of(System.getProperty("tests.s3source.region")))
+                .build();
+        bucket = System.getProperty("tests.s3source.bucket");
+        s3ObjectGenerator = new S3ObjectGenerator(s3Client, bucket);
+
+
+        sqsClient = SqsClient.builder()
+                .region(Region.of(System.getProperty("tests.s3source.region")))
+                .build();
+
+        s3SourceConfig = mock(S3SourceConfig.class);
+        buffer = mock(Buffer.class);
+        codec = mock(Codec.class);
+        pluginMetrics = mock(PluginMetrics.class);
+        s3Service = mock(S3Service.class);
+
+        final SqsOptions sqsOptions = mock(SqsOptions.class);
+        when(sqsOptions.getSqsUrl()).thenReturn(System.getProperty("tests.s3source.queue.url"));
+        when(sqsOptions.getVisibilityTimeout()).thenReturn(Duration.ofSeconds(60));
+        when(sqsOptions.getMaximumMessages()).thenReturn(10);
+        when(sqsOptions.getWaitTime()).thenReturn(Duration.ofSeconds(10));
+        when(s3SourceConfig.getSqsOptions()).thenReturn(sqsOptions);
+        when(s3SourceConfig.getOnErrorOption()).thenReturn(OnErrorOption.DELETE_MESSAGES);
+    }
+
+    private SqsWorker createObjectUnderTest() {
+        return new SqsWorker(sqsClient, s3Service, s3SourceConfig);
+    }
+
+    @AfterEach
+    void processRemainingMessages() {
+        final SqsWorker objectUnderTest = createObjectUnderTest();
+        int sqsMessagesProcessed;
+        do {
+            sqsMessagesProcessed = objectUnderTest.processSqsMessages();
+        }
+        while (sqsMessagesProcessed > 0);
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {1, 5})
+    void processSqsMessages_should_return_at_least_one_message(final int numberOfObjectsToWrite) throws IOException {
+        writeToS3(numberOfObjectsToWrite);
+
+        final SqsWorker objectUnderTest = createObjectUnderTest();
+        int sqsMessagesProcessed = objectUnderTest.processSqsMessages();
+
+        assertThat(sqsMessagesProcessed, greaterThanOrEqualTo(1));
+        assertThat(sqsMessagesProcessed, lessThanOrEqualTo(numberOfObjectsToWrite));
```

Comment:
We should be more exact here.

```
assertThat(sqsMessagesProcessed, equals(numberofObjectsToWrite));
```

Reply:
SQS doesn't guarantee to return same number of messages even if long polling and max messages is enabled. It will always be minimum of 1 and maximum of number of objects written to S3.

Reply:
As this is not fully obvious behavior, this may be a good place to leave some documentation explaining it.